docs/DeveloperGuide.md

```diff
@@ -4,9 +4,457 @@
+- `previous()`: This method moves the `startOfView` date to the start of the previous week by subtracting one week.
+
+The `WeekView` class also provides the following utility methods:
+
+- `getStartOfWeek()`: Returns the `startOfView` date, which represents the start of the current week.
+- `getDateForDay(int dayOfWeek)`: Returns the date for a specific day of the week based on the `startOfView` date. The `dayOfWeek` parameter is expected to be a value between 1 (Monday) and 7 (Sunday).
+
+### MonthView
+
+The `MonthView` class extends the `View` abstract class and provides an implementation for rendering and navigating the month view. It overrides the following methods from the `View` class:
+
+- `printView(TaskManager taskManager)`: This method first calculates the start of the month and the first day of the month that falls on a Sunday. It then prints the month header using the `UiRenderer.printWeekHeader` method with the `isMonthView` flag set to `true`. The method iterates over the weeks in the month and calls the `printWeek` method for each week.
+- `next()`: This method advances the `startOfView` date to the start of the next month.
+- `previous()`: This method moves the `startOfView` date to the start of the previous month.
+
+The `MonthView` class also provides the following private methods:
+
+- `printWeek(LocalDate current, TaskManager taskManager)`: This method prints a single week within the month view. It iterates over the days of the week and calls the `printDay` method for each day. It also prints the tasks for the week using the `printTasksForWeek` method.
+- `printDay(LocalDate current)`: This method prints the day number for a given date within the month view.
+- `getMaxTasksForWeek(LocalDate weekStart, TaskManager taskManager)`: This method calculates the maximum number of tasks for a given week by iterating over the days of the week and finding the maximum number of tasks for any day.
+- `printTasksForWeek(LocalDate weekStart, int maxTasks, TaskManager taskManager)`: This method prints the tasks for a given week within the month view. It iterates over the tasks and calls the `printTaskForDay` method for each task.
+- `printTaskForDay(List<Task> dayTasks, int taskIndex)`: This method prints a single task for a given day within the month view.
+
+### View Switching in Main
+
+The `Main` class is responsible for handling user input and dispatching commands to the appropriate components. It manages the switching between the week and month views based on the user's input.
+
+When the user enters the `month` command, the `Main` class toggles the `inMonthView` flag and calls the `printView` method of the `MonthView` instance, rendering the month view. Similarly, when the user enters the `week` command, the `Main` class sets the `inMonthView` flag to `false` and calls the `printView` method of the `WeekView` instance, rendering the week view.
```

Comment:
inconsistent formatting when compared to other sections of the developer's guide. Other sections utilised bullet points to convey the necessary information while this section conveyed all of its points in a paragraph.